### PR TITLE
Build overhaul - step two

### PR DESCRIPTION
**Pull request by markxnelson**
# Description

This PR continues the build cleanup work by introducing the following changes:

- Run the quality tools (fmt, lint, etc.) once from the project root
- Fix several issues in the tests package identified by quality tools
- Refactor shared makefile content into include files
- Fix the manifest goals which were broken by the previous PR
- Other miscellaneous changes (add TPL files to .gitignore, fix year in license file, etc.)

Contributes to VZ-1973

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [x] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/Makefile [hunk 1]
```diff
@@ -65,8 +68,6 @@ install-crds:
 uninstall-crds:
 	kustomize build config/crd | kubectl delete -f -
 
-.PHONY: check
-check: go-fmt go-vet go-ineffassign go-lint
```
Comment (pfmackin) [inline platform-operator/Makefile:69]: I use make check all the time.  Are we supposed to now do it from the top level?

Comment (markxnelson) [reply platform-operator/Makefile:69]: no it works exactly the same as before

File: make/generate.mk [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	./hack/update-codegen.sh "clusters:v1alpha1" "clusters" "boilerplate-clusters.go.txt"
+
+	# Re-generate operator.yaml using template yaml file
+	cat config/deploy/verrazzano-platform-operator.yaml | sed -e "s|IMAGE_NAME|$(shell grep --max-count=1 "image:" deploy/operator.yaml | awk '{ print $$2 }')|g" > deploy/operator.yaml
+	cat config/crd/bases/install.verrazzano.io_verrazzanos.yaml >> deploy/operator.yaml
+	cat config/crd/bases/clusters.verrazzano.io_verrazzanomanagedclusters.yaml >> deploy/operator.yaml
+
+# Generate manifests e.g. CRD, RBAC etc.
+.PHONY: application-manifests
+applicaiton-manifests: go-mod controller-gen
```
Comment (desagar) [inline make/generate.mk:20]: typo - referenced in the other Makefile with the right spelling of application but misspelled here

Comment (markxnelson) [reply make/generate.mk:20]: d'oh thanks :)

**Review by desagar — APPROVED** [review]

**Pull request merged**